### PR TITLE
Reorganize plugin development internal doc

### PR DESCRIPTION
**Pull request by lucie-dubrunfaut**
MON-16701

## Description

MON-16701 : Reorganization and update plugin development documentation

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [x] Documentation

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x (master)

### CODE REVIEW

File: doc/en/developer/guide.md [hunk 1]
```diff
@@ -1,268 +1,276 @@
...
 
-```perl
+The snmp_standard/mode exists since the beginning when SNMP monitoring was much more used than it is today. All the modes it contains use standard OIDs, which means that many plugins are relying on these when the manufacturer supports standard MIBs on their devices.
 
-  # ...
-  # Authors : <your name> <<your email>>
+<div id='architecture_plugin'/>
 
-```
+### 3. Plugin.mp file
```
Comment (garnier-quentin) [inline doc/en/developer/guide.md:160]: Plugin.pm

File: doc/en/developer/guide.md [hunk 2]
```diff
@@ -1,268 +1,276 @@
...
-
-        return $self;
-  }
-
+use strict;
+use warnings;
 ```
-
-Plugin version must be declared in the **new** constructor:
+Centreon librarie :
```
Comment (garnier-quentin) [inline doc/en/developer/guide.md:187]: Library

File: doc/en/developer/guide.md [hunk 3]
```diff
@@ -1,268 +1,276 @@
...
-
-  $self->{version} = '0.1';
-
+use base qw(**centreon_library**);
 ```
+There are five kinds of centreon libraries here :
+* centreon::plugins::script_simple : Previously the general use case if no custom is needed, more explainations [here]((#custom_mode_tuto) in this section.
+* centreon::plugins::script_custom : Need custom directory - More explainations [here]((#custom_mode_tuto) in this section.
+* centreon::plugins::script_snmp : If SNMP protocol is needed for this plugin
+* centreon::plugins::script_sql : If DB acess is needed for this plugin
```
Comment (garnier-quentin) [inline doc/en/developer/guide.md:196]: access

File: doc/en/developer/guide.md [hunk 4]
```diff
@@ -772,54 +822,51 @@ return_type  | Int (0 or 1) | 0       | Return a hash table with one level inste
...
-
-  my $results = $self->{snmp}->get_multiple_table(oids => [
-                                                        { oid => $aix_swap_pool, start => 1 },
-                                                        { oid => $oid_sysDescr },
-                                                  ]);
-
-  use Data::Dumper;
-  print Dumper($results);
+my $results = $self->{snmp}->get_multiple_table(oids => [
+                                                      { oid => $aix_swap_pool, start => 1 },
```
Comment (garnier-quentin) [inline doc/en/developer/guide.md:860]: start => 1 is not working (it's an suboid needed)

**Comment by omercier** [comment]
You must target `develop` branch, not `master` ;-)

**Review by YanMeddour — DISMISSED** [review]

File: doc/en/developer/guide.md (hunk 4 above)
Comment (lucie-dubrunfaut) [reply doc/en/developer/guide.md:860]: May I ask what you suggest to replace it ? 
_This part was legacy from 2020 tutorial._

File: .github/workflows/get-environment.yml [hunk 5]
```diff
@@ -22,6 +22,12 @@ jobs:
     steps:
       - id: get_environment
         run: |
+          if [[ -z "$GITHUB_HEAD_REF" ]]; then
+            BRANCHNAME="$GITHUB_REF_NAME"
+          else
+            BRANCHNAME="$GITHUB_HEAD_REF"
+          fi
+
```
Comment (garnier-quentin) [inline .github/workflows/get-environment.yml:30]: i'm not sure that file shoud be changed ?

File: doc/en/developer/tutorial-api.md [hunk 6]
```diff
@@ -0,0 +1,1116 @@
...
+    # Adding output structure to the object
+    $self->{output} = $options{output};
+    # Command line legacy from appsmetrics.pm in single mode
+    $self->{http} = centreon::plugins::http->new(%options);
+
+    return $self;
+}
+```
+
+#### set_options
```
Comment (lucie-dubrunfaut) [inline doc/en/developer/tutorial-api.md:1008]: This section should have more explanations about different functions but at this time some part remains unclear for me.

File: doc/en/developer/plugins_global.md [hunk 7]
```diff
@@ -0,0 +1,1075 @@
...
+To convert your plugin into Fatpack format follow this steps :
+
+Install libapp-fatpacker-perl
+```shell
+sudo apt install libapp-fatpacker-perl
+```
+
+Create a shell script ```plugin_generator.sh```
+
+```shell
```
Comment (lucie-dubrunfaut) [inline doc/en/developer/plugins_global.md:609]: this fatpack generator script may need some improvements

**Comment by lucie-dubrunfaut** [comment]
If you read this PR after all changes in October, please open guide.md and then navigate in the differents chapters.

File: doc/en/developer/plugins_advanced.md [hunk 8]
```diff
@@ -0,0 +1,1274 @@
...
+**Description**
+
+Add performance data to output (print it with **display** method).
+Performance data are displayed after '|'.
+
+**Parameters**
+
+| Parameter | Type   | Default | Description                            |
+|-----------|--------|---------|----------------------------------------|
+| label     | String |         | Label of the performance data.         |
```
Comment (omercier) [inline doc/en/developer/plugins_advanced.md:74]:
```suggestion
| label     | String |         | [legacy] Label of the performance data.         |
| nlabel     | String |         | Label of the performance data in a more standard and explicit fashion.         |
```

File: .github/workflows/get-environment.yml [hunk 9]
```diff
@@ -29,12 +29,6 @@ jobs:
 
       - id: get_environment
         run: |
-          if [[ -z "$GITHUB_HEAD_REF" ]]; then
-            BRANCHNAME="$GITHUB_REF_NAME"
-          else
-            BRANCHNAME="$GITHUB_HEAD_REF"
-          fi
-
```
Comment (sc979) [inline .github/workflows/get-environment.yml:37]: Kindly, could you restore this part ?
Until then, $BRANCHNAME will remain empty in the case below

**Review by sc979 — CHANGES_REQUESTED** [review]

File: .github/workflows/get-environment.yml (hunk 9 above)
Comment (lucie-dubrunfaut) [reply .github/workflows/get-environment.yml]: Done :)

**Review by sc979 — APPROVED** [review]
LGTM on pipeline part

**Review by omercier — APPROVED** [review]

**Review by bouda1 — APPROVED** [review]

**Review by bouda1 — APPROVED** [review]

**Review by sdepassio — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**